spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/CachingClientConnectionFactoryTests.java

```diff
@@ -505,17 +506,13 @@ private void testCloseOnTimeoutGuts(AbstractClientConnectionFactory cf) throws E
 	@Test
 	public void testCachedFailover() throws Exception {
 		// Failover
-		AbstractClientConnectionFactory factory1 = mock(AbstractClientConnectionFactory.class);
-		AbstractClientConnectionFactory factory2 = mock(AbstractClientConnectionFactory.class);
+		TcpConnectionSupport mockConn1 = makeMockConnection();
+		TcpConnectionSupport mockConn2 = makeMockConnection();
+		AbstractClientConnectionFactory factory1 = FailoverClientConnectionFactoryTests.makeMockactory(mockConn1);
+		AbstractClientConnectionFactory factory2 = FailoverClientConnectionFactoryTests.makeMockactory(mockConn2);
```

Comment:
First of all there is a typo in the method name.
Secondly I'd prefer to mix test classes with each other.
Even if right now they have similar logic for mock it might be hard in the future to refactor them.
My point is that those tests are essentially closed boxed and it is better to design them without tangling with other test classes.
It might look right now as a readability improvement, but it will be harder to support them individually in the future when the logic one of them might change.

In the end these are just tests, so might not worse pursuing such a refactoring if you cannot find a solution to not tangle test classes with each other.